packages/@lwc/engine-core/package.json

```diff
@@ -17,7 +17,7 @@
     "license": "MIT",
     "scripts": {
         "clean": "rm -rf dist/ types/",
-        "build": "tsc --emitDeclarationOnly && rollup --config scripts/rollup.config.js"
+        "build": "rollup --config scripts/rollup.config.js && rm -fr types && mv dist/types types"
```

Comment:
I couldn't think of a better way to do this. Otherwise the `types` files get written to `./dist/types` rather than `./types`, which would be a breaking change.

`'../types'` does not work; it complains about relative paths.

Reply:
I see the problem. I'll double-check if there's some way to integrate the `mv`/`rm/` nastiness into Rollup itself.

I don't use `yarn build --watch` but I can see how it would be valuable.

Reply:
It looks like `@rollup/plugin-typescript` [explicitly tries](https://github.com/rollup/plugins/blob/e32a9e97a4bb4242ae597fb08177bb320db1b595/packages/typescript/test/test.js#L52) to include `types` only in the `dist` directory.

I can't find a workaround except by using a small Rollup plugin:

```js
            {
                id: 'write-dist-and-types',
                generateBundle(options, bundle) {
                    for (const [ id, descriptor ] of Object.entries(bundle)) {
                        const directory = id.endsWith('.d.ts') ? 'types': 'dist';
                        descriptor.fileName = path.join(directory, descriptor.fileName);
                    }
                    return null
                }
            }
```

It's a bit ugly, but I'll go ahead with it since it unbreaks `--watch` and allows us to remove the `mv` and `rm` stuff.

Reply:
OK, got it working, and the `dist/` and `types/` output is unchanged.